File: Data Structures & Algorithms/maximum-subarray/submission-1.java
class Solution {
    public int maxSubArray(int[] nums) {
        int right = 0;
        int left = 0;
        int max = nums[0];
        int total = 0;
        while (right < nums.length) {
            if (total < 0) {
                left = right;
                total = 0;
            }
            total += nums[right];
            max = Math.max(max, total);
            right++;

        }
        
        return max;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        total = 1
        zeroes = 0
        for num in nums:
            if num == 0:
                zeroes += 1
            else:
                total *= num
        if zeroes > 1:
            return [0] * len(nums)
        
        res = []
        for num in nums:
            if zeroes:
                if num == 0:
                    res.append(total)
                else:
                    res.append(0)
            else:
                res.append(total//num)
        return res